src/frontend/mame/ui/viewgfx.cpp

```diff
@@ -1251,7 +1281,7 @@ static void tilemap_update_bitmap(running_machine &machine, ui_gfx_state &state,
 		std::swap(width, height);
 
 	// realloc the bitmap if it is too small
-	if (state.bitmap == nullptr || state.texture == nullptr || state.bitmap->width() != width || state.bitmap->height() != height)
+	if (state.bitmap_dirty || state.bitmap == nullptr || state.texture == nullptr || state.bitmap->width() != width || state.bitmap->height() != height)
```

Comment:
I'm doing this as pull request mainly because of this. This causes the system to completely realloc the render texture on all dirty changes (as opposed to only on size changes before). I had to add this to get it to render the tilemaps correctly. Probably this can be done in a better way?

Note: It doesn't have a performance impact because this only applies to dirty changes from the tilemap viewer, not from the actual driver.